vllm/model_executor/models/chatglm.py

```diff
@@ -306,26 +442,67 @@ def __init__(
                                            config.hidden_size,
                                            quant_config=quant_config)
 
+        vision_config_flag = getattr(config, 'vision_config', None)
+        if vision_config_flag is not None:
+            self.vision_config = Namespace(**config.vision_config)
+            self.vision = EVA2CLIPModel(self.config, quant_config)
+        else:
+            self.vision = None
+
     def forward(
         self,
         input_ids: torch.Tensor,
-        position_ids: torch.Tensor,
+        positions: torch.Tensor,
         kv_caches: List[torch.Tensor],
         attn_metadata: AttentionMetadata,
+        intermediate_tensors: Optional[IntermediateTensors] = None,
+        **kwargs: object,
     ) -> torch.Tensor:
+
         inputs_embeds = self.embedding(input_ids)
+        pixel_values = kwargs.pop("image_embeds", None)
+        if pixel_values is not None and self.vision is not None:
+
+            if isinstance(pixel_values, torch.Tensor):
+                if pixel_values.ndim == 2:
+                    pixel_values = pixel_values
+                else:
+                    pixel_values = torch.concat(list(pixel_values))
+            elif isinstance(pixel_values, list):
+                return torch.concat(pixel_values)
```

Comment:
Is there a reason pixel values get returned directly here if it's a list instead of merging the multimodal embeddings & running the encoder?